### PR TITLE
Add num.parse and support 0x/0b prefixes for int.parse.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/core/numbers.toit [hunk 1]
```diff
@@ -535,6 +537,26 @@ abstract class num implements Comparable:
...
 
+  /**
+  Parses the given $data as a number.
+
+  Tries to parse the $data as an integer first, and if that fails, as a float.
+
+  See $int.parse and $float.parse.
+  */
+  static parse data/io.Data -> num:
+    return int.parse data --on-error=: return float.parse data --on-error=: throw PARSE-ERR_
```
Comment (kasperl) [inline lib/core/numbers.toit:548]: Maybe format this like the variant below?

File: tests/number-test.toit [hunk 2]
```diff
@@ -113,6 +117,16 @@ test-round:
   expect-number-out-of-range: float.INFINITY.round
   expect-number-out-of-range: float.NAN.round
 
+test-parse-number:
+  expect-identical 0 (num.parse "0")
+  expect-identical 0 (num.parse "-0")
+  expect-identical 0.0 (num.parse "0.0")
+
+  expect-identical int.MAX (num.parse "0b111111111111111111111111111111111111111111111111111111111111111")
+  expect-identical 9223372036854775808.0 (num.parse "9223372036854775808")
```
Comment (kasperl) [inline tests/number-test.toit:126]: Maybe add a comment that this is int.MAX + 1 as a float (if I am not mistaken)?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/number-test.toit (hunk 2 above)
Comment (floitsch) [reply tests/number-test.toit:127]: done.

**Pull request merged**